octant doesn't build on OS-X

I'm getting:

/usr/local/bin/gfortran -Wall -g -arch x86_64 build/temp.macosx-10.5-x86_64-2.7/build/src.macosx-10.5-x86_64-2.7/_isomodule.o build/temp.macosx-10.5-x86_64-2.7/build/src.macosx-10.5-x86_64-2.7/fortranobject.o build/temp.macosx-10.5-x86_64-2.7/octant/src/iso.o -L/usr/local/gfortran/lib/gcc/i686-apple-darwin10/4.6.1/x86_64 -L/Users/chris.barker/anaconda/envs/_build/lib -lgfortran -o build/lib.macosx-10.5-x86_64-2.7/octant/_iso.so
gfortran: error: x86_64: No such file or directory
gfortran: error: unrecognized option ‘-arch’
gfortran: error: x86_64: No such file or directory
gfortran: error: unrecognized option ‘-arch’


I'm pretty sure the issue here is that python's distutils puts in the -arch flags (so that it can do universal builds in some cases). but the gfortran I've got doesn't recognise those flags. Honestly Im not sure where I got that gfortran -- though if I"ve done anyting with it -- it would be comiling fortran python extensions, so this is a bit odd.

But what should we use for fortran for Anaconda in OS-X? The compilers Apple ships don't have it.

OK: trying octant again with the homebrew gfortran:

$ gfortran --version
GNU Fortran (Homebrew gcc 4.9.2_1) 4.9.2
Copyright (C) 2014 Free Software Foundation, Inc.

But it bombs out -- looks like it's having trouble with the fidning teh python libs. a bunch of errors like:

Undefined symbols for architecture x86_64:
  "_PyArg_ParseTupleAndKeywords", referenced from:
      _f2py_rout__iso_integrate in _isomodule.o
      _f2py_rout__iso_surface in _isomodule.o
      _f2py_rout__iso_zslice in _isomodule.o

and lots more -- but this is all standard python stuff.

More later.....

--- a/octant/build.sh
+++ b/octant/build.sh
@@ -1,9 +1,10 @@
 #!/bin/bash
 
+if [ `uname` == Darwin ]; then
+    export CFLAGS="-arch x86_64"
+    export FFLAGS="-static -ff2c -arch x86_64"
+    export LDFLAGS="-Wall -undefined dynamic_lookup -bundle -arch x86_64"
+    export LDFLAGS="-Wl,-search_paths_first -L$(pwd) $LDFLAGS"
+fi
+
 $PYTHON setup.py install
-
-# Add more build steps here, if they are necessary.
-
-# See
-# http://docs.continuum.io/conda/build.html
-# for a list of environment variables that are set during the build process.

